# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/batch.rs
# Commit: 0edb6b6f17b2
# Full Hash: 0edb6b6f17b24c737d053df19730e9c9e61e17ca
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r ee0de0eac18d -r 0edb6b6f17b2 gfx/wr/webrender/src/batch.rs
--- a/gfx/wr/webrender/src/batch.rs	Thu Dec 17 22:26:08 2020 +0000
+++ b/gfx/wr/webrender/src/batch.rs	Thu Dec 17 22:26:10 2020 +0000
@@ -1611,7 +1611,7 @@
                                         ).unwrap();
 
                                         let kind = BatchKind::Brush(
-                                            BrushBatchKind::Image(ImageBufferKind::Texture2DArray)
+                                            BrushBatchKind::Image(ImageBufferKind::Texture2D)
                                         );
                                         let (uv_rect_address, textures) = render_tasks.resolve_surface(
                                             surface_task.expect("bug: surface must be allocated by now"),
@@ -1658,7 +1658,7 @@
 
                                         // The shadows and the content get drawn as a brush image.
                                         let kind = BatchKind::Brush(
-                                            BrushBatchKind::Image(ImageBufferKind::Texture2DArray),
+                                            BrushBatchKind::Image(ImageBufferKind::Texture2D),
                                         );
 
                                         // Gets the saved render task ID of the content, which is
@@ -1669,7 +1669,6 @@
                                             let texture_id = secondary_task.get_target_texture();
                                             TextureSource::TextureCache(
                                                 texture_id,
-                                                ImageBufferKind::Texture2DArray,
                                                 Swizzle::default(),
                                             )
                                         };
@@ -1948,7 +1947,7 @@
                                 );
                                 let key = BatchKey::new(
                                     BatchKind::Brush(
-                                        BrushBatchKind::Image(ImageBufferKind::Texture2DArray),
+                                        BrushBatchKind::Image(ImageBufferKind::Texture2D),
                                     ),
                                     BlendMode::Advanced(mode),
                                     textures,
@@ -2008,12 +2007,10 @@
                                             colors: [
                                                 TextureSource::TextureCache(
                                                     color0,
-                                                    ImageBufferKind::Texture2DArray,
                                                     Swizzle::default(),
                                                 ),
                                                 TextureSource::TextureCache(
                                                     color1,
-                                                    ImageBufferKind::Texture2DArray,
                                                     Swizzle::default(),
                                                 ),
                                                 TextureSource::Invalid,
@@ -2056,7 +2053,6 @@
                                     colors: [
                                         TextureSource::TextureCache(
                                             texture_id,
-                                            ImageBufferKind::Texture2DArray,
                                             Swizzle::default(),
                                         ),
                                         TextureSource::Invalid,
@@ -2064,7 +2060,7 @@
                                     ],
                                 };
                                 let batch_params = BrushBatchParameters::shared(
-                                    BrushBatchKind::Image(ImageBufferKind::Texture2DArray),
+                                    BrushBatchKind::Image(ImageBufferKind::Texture2D),
                                     textures,
                                     ImageBrushData {
                                         color_mode: ShaderColorMode::Image,
@@ -2130,7 +2126,7 @@
                                 ).unwrap();
 
                                 let kind = BatchKind::Brush(
-                                    BrushBatchKind::Image(ImageBufferKind::Texture2DArray)
+                                    BrushBatchKind::Image(ImageBufferKind::Texture2D)
                                 );
                                 let (uv_rect_address, textures) = render_tasks.resolve_surface(
                                     surface_task.expect("bug: surface must be allocated by now"),
@@ -2987,7 +2983,7 @@
                 );
 
                 let batch_key = BatchKey::new(
-                    BatchKind::Brush(BrushBatchKind::Image(ImageBufferKind::Texture2DArray)),
+                    BatchKind::Brush(BrushBatchKind::Image(ImageBufferKind::Texture2D)),
                     BlendMode::PremultipliedAlpha,
                     textures,
                 );
@@ -3349,7 +3345,6 @@
             BatchTextures::prim_textured(
                 TextureSource::TextureCache(
                     task.get_target_texture(),
-                    ImageBufferKind::Texture2DArray,
                     Swizzle::default(),
                 ),
                 clip_mask,
@@ -3832,7 +3827,6 @@
                     task_id.into(),
                     TextureSource::TextureCache(
                         render_tasks[task_id].get_target_texture(),
-                        ImageBufferKind::Texture2DArray,
                         Swizzle::default(),
                     )
                 ))